src/providers/vk.js

```diff
@@ -0,0 +1,30 @@
+export default (options) => {
+  const apiVersion = '5.126' // https://vk.com/dev/versions
+
+  return {
+    id: 'vk',
+    name: 'vk.com',
+    type: 'oauth',
+    version: '2.0',
+    scope: 'email',
+    params: {
+      grant_type: 'authorization_code'
+    },
+    accessTokenUrl: `https://oauth.vk.com/access_token?v=${apiVersion}`,
+    requestTokenUrl: `https://oauth.vk.com/access_token?v=${apiVersion}`,
+    authorizationUrl:
+      `https://oauth.vk.com/authorize?response_type=code&v=${apiVersion}`,
+    profileUrl: `https://api.vk.com/method/users.get?fields=photo_100&v=${apiVersion}`,
+    profile: (result) => {
+      const profile = result.response?.[0]
```

Comment:
I like the use of optional chaining here, but you should probably add a fallback value with `?? {}`, otherwise property access on `profile` will fail if profile is undefined.

Could you tell me in which case the response won't include any profile information? It seems strange you need to check for existence at all.

Reply:
You're right. Added fallback, previously there was just an early return with `if (!profile) return {}`.

Response will contain `error` object with error code and message instead of users info, for example, if one omits `v=${apiVersion}` param from url.